### ISSUE TITLE
Problem while integration flutter AAR in android app

### ISSUE THREAD

**Issue by charbel-elhajj** [flutter/flutter#114695]
Hello everyone!

I am working on a project where I'm integrating a flutter package I'm developing in an app built in another multi-platform development framework. To do so, I am going through the native layer, so it is as if I would be integrating my flutter package into iOS and Android native apps. Here is the [guide](https://docs.flutter.dev/development/add-to-app) to do so. For android, I am using the AAR approach.

While shifting to the latest stable flutter versions (3.3.7) I noticed that the output of the command `flutter build aar` has changed and is now different than the documentation. It has become : 

```
  1. Open <host>/app/build.gradle
  2. Ensure you have the repositories configured, otherwise add them:

      String storageUrl = System.env.FLUTTER_STORAGE_BASE_URL ?: "https://storage.googleapis.com"
      repositories {
        maven {
            url 'path_to_my_package/build/host/outputs/repo'
        }
        maven {
            url "$storageUrl/download.flutter.io"
        }
      }

  3. Make the host app depend on the Flutter module:

    dependencies {
      debugImplementation 'my_package:flutter:1.0:debug'
      profileImplementation 'my_package:flutter:1.0:profile'
      releaseImplementation 'my_package:flutter:1.0:release'
    }


  4. Add the `profile` build type:

    android {
      buildTypes {
        profile {
          initWith debug
        }
      }
    }

```

And when building the app for android, I get the following error

```
Could not find flutter-1.0-release.jar
```

Note: I was using flutter 3.0.0 before that.

Did anything change in the way we should add Flutter to an android app through an AAR ?

**Comment by danagbemava-nc** [comment]
Hi @charbel-elhajj, can you share the output of `flutter build aar` from the version you were on prior to 3.3.7? 

Please also share the full logs of when you try to build the android app.

Thank you

**Comment by charbel-elhajj** [comment]
Here is the output when using Flutter 3.0.0

```
💪 Building with sound null safety 💪

Running Gradle task 'assembleAarDebug'...                         125.5s
✓ Built build/host/outputs/repo.
Running Gradle task 'assembleAarProfile'...                       116.7s
✓ Built build/host/outputs/repo.
Running Gradle task 'assembleAarRelease'...                        99.5s
✓ Built build/host/outputs/repo.

Consuming the Module
  1. Open <host>/app/build.gradle
  2. Ensure you have the repositories configured, otherwise add them:

      String storageUrl = System.env.FLUTTER_STORAGE_BASE_URL ?: "https://storage.googleapis.com"
      repositories {
        maven {
            url 'path_to_my_package/build/host/outputs/repo'
        }
        maven {
            url "$storageUrl/download.flutter.io"
        }
      }

  3. Make the host app depend on the Flutter module:

    dependencies {
      **debugImplementation 'my_package:flutter_debug:1.0'
      profileImplementation 'my_package:flutter_profile:1.0'
      releaseImplementation 'my_package:flutter_release:1.0'**
    }


  4. Add the `profile` build type:

    android {
      buildTypes {
        profile {
          initWith debug
        }
      }
    }

```

App build output with flutter 3.3.7 built aar (it is a cordova application built on android) :
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugKotlin'.
> Error while evaluating property 'filteredArgumentsMap' of task ':app:compileDebugKotlin'
   > Could not resolve all files for configuration ':app:debugCompileClasspath'.
      > Failed to transform flutter-1.0-release.jar (my_package:flutter:1.0) to match attributes {artifactType=android-classes-jar, org.gradle.libraryelements=jar, org.gradle.status=release, org.gradle.usage=java-runtime}.
         > Could not find flutter-1.0-release.jar (my_package:flutter:1.0).
           Searched in the following locations:
               file:path_to_app/cordova_host_application/platforms/android/app/src/main/kotlin/Flutter/repo/com/my_package/flutter/1.0/flutter-1.0-release.jar

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.1.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 42s
24 actionable tasks: 1 executed, 23 up-to-date
Command failed with exit code 1: path_to_app/cordova_host_application/platforms/android/gradlew cdvBuildDebug -b path_to_app/cordova_host_application/platforms/android/build.gradle
``` 

Note that there is a `flutter-1.0-release.aar` in the location where they are searching and there was never a jar file built by flutter (neither on 3.0.0 nor on 3.3.7), only AARs, but this problem appeared only after the version upgrade.

**Comment by danagbemava-nc** [comment]
Hi @charbel-elhajj, thanks for the info. I believe this is a duplicate of https://github.com/flutter/flutter/issues/112380 albeit with different files throwing the error, as such I'll be closing this issue in favor of the said issue. 

Kindly follow up there for updates. 

If you disagree, kindly leave a comment below and I will reopen this. 

Thank you
